docker/cloudbuild.yaml

```diff
@@ -0,0 +1,35 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 600s
+# this prevents errors if you don't use both _GIT_TAG and _PULL_BASE_REF,
+# or any new substitutions added in the future.
+options:
+  substitution_option: ALLOW_LOOSE
+  # this is the machine type used for kubernetes releases.
+  # See https://cloud.google.com/build/docs/speeding-up-builds#using_custom_virtual_machine_sizes
+  # job builds a multi-arch docker image for amd64,arm64
+  machineType: 'N1_HIGHCPU_8'
+steps:
+  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20240718-5ef92b5c36'
+    entrypoint: bash
+    dir: ./docker
+    env:
+    - TAG=$_GIT_TAG
+    - BASE_REF=$_PULL_BASE_REF
+    - REGISTRY=us-central1-docker.pkg.dev/k8s-staging-images/secrets-store-sync
```

Comment:
@upodroid Is this the registry we need to use for cloudbuild config?

Reply:
@upodroid Any thoughts on this?